src/main/java/seedu/address/logic/commands/UndoCommand.java

```diff
@@ -0,0 +1,36 @@
+package seedu.address.logic.commands;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.state.State;
+
+/**
+ * Undoes the user operations on the address book. This command only applies to the commands
+ * that make changes to the address book, such as add, delete, edit etc.
+ */
+public class UndoCommand extends Command {
+
+    public static final String COMMAND_WORD = "undo";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Undoes the changes to the address book. ";
+
+    public static final String MESSAGE_SUCCESS = "Undo operation successful!";
+
+    public static final String MESSAGE_FAILED = "Cannot undo!";
```

Comment:
I feel that the exclamation marks are unnecessary and do not fit the style of the app responses. Is there a reason for them?

Reply:
I added it instinctively.